GTViewController/Classes/Nav Classes/GTFollowupViewController.m

```diff
@@ -100,4 +146,146 @@ - (void)transitionToThankYou {
         [self.view bringSubviewToFront:self.followupThankYouView];
     }];
 }
+
+
+#pragma mark Animation methods to prevent text field from being hidden
+
+-(void)keyboardWillShow:(NSNotification *) notification {
+    if (self.view.frame.origin.y < 0) {
+        return;
+    }
+    
+    [self moveView:notification];
+}
+
+-(void)keyboardWillHide:(NSNotification *) notification {
+    
+    if (self.view.frame.origin.y >= 0) {
+        return;
+    }
+    
+    [self resetViewPosition:notification];
+}
+
+
+//method to move the view up when the keyboard would cover the "active" textField
+-(void)moveView:(NSNotification *)notification {
+    NSDictionary *userInfo = notification.userInfo;
+    
+    CGRect keyboardEndFrame;
+    
+    [[userInfo objectForKey:UIKeyboardFrameEndUserInfoKey] getValue:&keyboardEndFrame];
+    
+    CGRect keyboardFrame = [self.view convertRect:keyboardEndFrame toView:nil];
+    CGRect inputFieldViewFrame = self.activeField.superview.frame;
+    
+    CGRect unobscuredByKeyboardFrame = self.view.frame;
+    unobscuredByKeyboardFrame.size.height -= keyboardFrame.size.height;
+    
+    // offset the height of the inputFieldView, b/c it is a label stacked on an input
+    // if just the frame is considered, then the label could show while the input is hidden
+    unobscuredByKeyboardFrame.size.height -= inputFieldViewFrame.size.height;
+    
+    if (CGRectContainsPoint(unobscuredByKeyboardFrame, inputFieldViewFrame.origin) ) {
+        return;
+    }
+
+    CGFloat viewVerticalDelta = (inputFieldViewFrame.origin.y + inputFieldViewFrame.size.height) - unobscuredByKeyboardFrame.size.height;
+    UIViewAnimationCurve animationCurve;
+    NSTimeInterval animationDuration;
+    
+    [[userInfo objectForKey:UIKeyboardAnimationCurveUserInfoKey] getValue:&animationCurve];
+    [[userInfo objectForKey:UIKeyboardAnimationDurationUserInfoKey] getValue:&animationDuration];
+
+    [self animateViewWithVerticalDelta:viewVerticalDelta
+                     animationDuration:animationDuration
+                        animationCurve:animationCurve];
+}
+
+
+- (void) animateViewWithVerticalDelta:(CGFloat)verticalDelta animationDuration:(NSTimeInterval) animationDuration animationCurve:(UIViewAnimationCurve)animationCurve {
+    // Get animation info from userInfo
+    CGRect newViewFrame = self.view.frame;
+    
+    [UIView beginAnimations:nil context:nil];
+    [UIView setAnimationDuration:animationDuration];
+    [UIView setAnimationCurve:animationCurve];
+    
+    newViewFrame.origin.y -= verticalDelta;
```

Comment:
what was the code you used instead of this? Those methods create new CGRect structures, they don't act on self.view.frame itself. You will need to assign the newly created frame back to self.view.frame
That is the most common issue that comes up with this kind of animation.